File: Projects/codechef/SMRSTR.java
import java.io.*;
import java.util.*;

class SMRSTR {
	public static void main (String[] args) throws IOException {
		Scanner in = new Scanner(System.in);
		int t = in.nextInt();;
		while (t-- > 0) {
			int n = in.nextInt();
			int q = in.nextInt();
			long p = 1;
			while (n-- > 0 && p <= (int) 1e9)
				p *= in.nextInt();
			in.nextLine();
			while (q-- > 0)
				System.out.printf("%d ", in.nextInt() / p);
			System.out.println();
		}
	}
}

File: Projects/codechef/SMRSTR.kt
import java.io.*
import java.util.*

internal object SMRSTR {
    @Throws(IOException::class)
    fun main(args: Array<String>) {
        val stdin = Scanner(System.`in`)
        var t = stdin.nextInt()
        while (t-- > 0) {
            var n = stdin.nextInt()
            var q = stdin.nextInt()
            var p: Long = 1
            while (n-- > 0 && p <= 1e9.toInt())
                p *= stdin.nextInt()
            stdin.nextLine()
            while (q-- > 0)
                System.out.printf("%d ", stdin.nextInt() / p)
            System.out.println()
        }
    }
}
